subsys/fs/Kconfig.littlefs

```diff
@@ -104,4 +104,24 @@ config FS_LITTLEFS_HEAP_PER_ALLOC_OVERHEAD_SIZE
 
 endif # FS_LITTLEFS_FC_HEAP_SIZE <= 0
 
+config LFS_READONLY
+	bool "Enable readonly little fs"
+	default 0
+	help
+	  Enable readonly littlefs
+
+config LFS_CFG
+	bool "Enable little fs custom config"
+	default 0
+	help
+	  Enable lfs custom config file. This flag will allow CUSTOM_LFS_CONFIG_FILE.
+
+config CUSTOM_LFS_CONFIG_FILE
+	string "Custom LFS configuration file" if LFS_CFG
+	default "littlefs_util.h"
+	help
+	  Allow user defined input file for the LFS_CONFIG setting. The default config
+	  file can be tweaked with Kconfig to optimize resources with lfs library.
+	  You can specify the actual configuration file using the LFS_CONFIG.
```

Comment:
we already have Kconfig for configuring LittleFS, so why do we need a second way of configuring LittleFS.

If users also have to consider a little fs header file for configuration then which configuration takes precedence Kconfig / header file.

How are the settings in the config header documented ?

When including files through Kconfig, then what will be the include path used ?
And what about absolute paths ?

Are there any settings in the config header that has dependencies to other settings ?
Kconfig allows dependencies to be described.